mcs/class/corlib/Test/System.Reflection/AssemblyTest.cs

```diff
@@ -182,7 +182,10 @@ public void GetEntryAssembly ()
 			// note: only available in default appdomain
 			// http://weblogs.asp.net/asanto/archive/2003/09/08/26710.aspx
 			// Not sure we should emulate this behavior.
-#if !MONODROID
+#if __WATCHOS__
+			Assert.IsNull (Assembly.GetEntryAssembly (), "GetEntryAssembly");
+			Assert.IsTrue (AppDomain.CurrentDomain.IsDefaultAppDomain (), "!default appdomain");
+#elif !MONODROID
```

Comment:
why not just `#if !MONODROID && !__WATCHOS__` as the #else is the same as this code here?